sdk/storage/azure_storage_blob/src/clients/blob_container_client.rs

```diff
@@ -59,18 +60,39 @@ impl BlobContainerClient {
 
         let client = GeneratedBlobContainerClient::new(
             endpoint,
-            credential,
+            credential.clone(),
             container_name.clone(),
             Some(options),
         )?;
 
         Ok(Self {
             endpoint: endpoint.parse()?,
             container_name,
+            credential,
             client,
         })
     }
 
+    /// Returns a new instance of BlobClient.
+    ///
+    /// # Arguments
+    ///
+    /// * `blob_name` - The name of the blob.
+    /// * `options` - Optional configuration for the client.
+    pub fn get_blob_client(
```

Comment:
Naming nit: Calling this function "get_xxx" implies that it is retrieving an existing something and not that it is creating something new. I recognize that the equivalent function is called `getXxx` in all the P0 languages, and C++, but in Go it is `NewXxx` so there is at least a convention in Go of calling "a function which creates a Foo" with "NewFoo".

Reply:
This a good callout, and I agree that `get_xxx` implies fetching something rather than new-ing something up. I wonder if this convention holds up in Rust, or would it be odd/confusing to have `new_blob_client` inside of `BlobContainerClient`? 

One thing to also note is that it seems like customers are used to `getXxx` from P0 languages but I am okay with cx having to read the documentation a little harder to find that `newXxx` is equivalent if that design is objectively better.
cc: @heaths @jalauzon-msft